internal/pubsub/pubsub.go

```diff
@@ -257,18 +257,34 @@ func (s *Subscription) As(i interface{}) bool {
 // It is for use by provider implementations.
 func NewSubscription(d driver.Subscription) *Subscription {
 	callCtx, cancel := context.WithCancel(context.Background())
-	handler := func(items interface{}) error {
-		ids := items.([]driver.AckID)
-		// TODO: Consider providing a way to stop this call. See #766.
-		return retry.Call(callCtx, gax.Backoff{}, d.IsRetryable, func() error {
-			return d.SendAcks(callCtx, ids)
-		})
-	}
 	const maxHandlers = 1
-	ab := batcher.New(reflect.TypeOf([]driver.AckID{}).Elem(), maxHandlers, handler)
+	handler := MakeAckBatchHandler(callCtx, d)
+	ackBatcher := batcher.New(reflect.TypeOf([]driver.AckID{}).Elem(), maxHandlers, handler)
+	return NewSubscriptionWithBatcher(d, ackBatcher, cancel)
+}
+
+// NewSubscriptionWithBatcher creates a Subscription from a driver.Subscription
+// and a batcher that sends batches of acks to the provider. Typically the
+// batcher is constructed using the result of MakeAckBatchHandler.
+// NewSubscriptionWithBatcher is for use by provider implementations.
+func NewSubscriptionWithBatcher(d driver.Subscription, ab driver.Batcher, cancel func()) *Subscription {
```

Comment:
Another way to do this (not sure if it's better, but consider it):

```
func NewSubscription(d driver.Subscription, makeBatcher func(ctx context.Context, handler func(interface{}) error)) {
  callCtx, cancel := context.WithCancel(context.Background())
  handler := /* handler code inline here */
  batcher := makeBatcher(callCtx, handler)
  return &Subscription{...}
}
```

and then have a `DefaultBatcher` function that provider implementations can pass to as `makeBatcher`. Tests would pass something like `makeTestBatcher`.

Reply:
This avoids the awkwardness of exposing the `MakeAckBatchHandler` function, but requiring the caller to pass in a `driver.Batcher`, so they have to connect those dots.

Reply:
Done.